saleor/product/tasks.py

```diff
@@ -100,6 +101,16 @@ def update_products_discounted_prices_of_promotion_task(promotion_pk: UUID):
     update_discounted_prices_task.delay(list(products.values_list("id", flat=True)))
 
 
+@app.task
+def mark_promotion_rules_to_recalculate_variants(promotion_pk: UUID):
```

Comment:
Do we need to have this as celery task? It looks like you're calling it as normal function.